src/cargo/ops/cargo_package.rs

```diff
@@ -405,6 +423,35 @@ fn warn_on_nonexistent_file(
     ))
 }
 
+fn error_custom_build_file_not_in_package(
+    pkg: &Package,
+    path: &Path,
+    target: &Target,
+    ws: &Workspace<'_>,
+) -> CargoResult<()> {
+    let tip = {
+        if path.is_file() {
+            format!("the source file of {:?} target `{}` doesn't appear to be a path inside of the package.\n\
+            It is at `{}`, whereas the root the package is `{}`.\n",
+            target.kind(), target.name(), path.display(), pkg.root().display()
+            )
+        } else {
+            format!(
+                "the source file of {:?} target `{}` doesn't appear to exist.\n",
+                target.kind(),
+                target.name()
+            )
```

Comment:
The error message shows "the source file of \"custom-build\" target `build-script-custom_build`", which does look correct to me. Maybe this?

```suggestion
            format!(
                "the source file of doesn't appear to exist.\n",
                target.description_named(),
            )
```

BTW I would suggest extract `target.description_named()` as a variable so that each branch can just use identifier in the format args position.